dbms/src/WindowFunctions/IWindowFunction.cpp

```diff
@@ -31,6 +32,29 @@ namespace ErrorCodes
 extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
 } // namespace ErrorCodes
 
+namespace
+{
+bool isFrameValid(RowNumber frame_start, RowNumber frame_end, RowNumber partition_start, RowNumber partition_end)
+{
+    return !((partition_end <= frame_start) || (frame_end <= partition_start));
+}
+
+bool defaultCheckFrameValidAndHandle(WindowTransformAction & action, size_t function_index)
+{
+    if (isFrameValid(action.frame_start, action.frame_end, action.partition_start, action.partition_end))
+        return true;
+
+    IColumn & to = *action.blockAt(action.current_row).output_columns[function_index];
+    static Field null_field;
+    to.insert(null_field);
+    return false;
+}
+} // namespace
+
+// In window function, some of the functions must return nullable column.
+// Return nullable column functions: FIRST_VALUE, LAST_VALUE, LAG, LEAD, NTH_VALUE, NTILE and all aggregation functions
```

Comment:
This comment is contradictory with the comment here https://github.com/pingcap/tiflash/pull/7939/files#diff-0f7c99237fd95a3f57b8eedb9e90e2f3b48f44d9abc8af6b0dcef0960d5ea37aR29 for LEAD/LAG

Reply:
> This comment is contradictory with the comment here https://github.com/pingcap/tiflash/pull/7939/files#diff-0f7c99237fd95a3f57b8eedb9e90e2f3b48f44d9abc8af6b0dcef0960d5ea37aR29 for LEAD/LAG

okk, I will refine it.